File: app/src/main/java/com/leodeleon/popmovies/feature/view/MoviePagerFragment.java
package com.leodeleon.popmovies.feature.view;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import butterknife.BindString;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import com.astuetz.PagerSlidingTabStrip;
import com.leodeleon.popmovies.R;
import com.leodeleon.popmovies.feature.MainActivity;

import static com.leodeleon.popmovies.feature.view.MoviesFragment.POSITION_POPULAR;
import static com.leodeleon.popmovies.feature.view.MoviesFragment.POSITION_RATED;

public class MoviePagerFragment extends Fragment {


  private static final int PAGE_GOUNT = 3;
  @BindView(R.id.viewpager) ViewPager mViewPager;
  @BindView(R.id.toolbar) Toolbar mToolbar;
  @BindView(R.id.tabs) PagerSlidingTabStrip mTabs;

  @BindString(R.string.popular) String popular;
  @BindString(R.string.sort_rated) String rated;
  @BindString(R.string.favorites) String favorites;

  private PagerAdapter adapter;
  private Unbinder unbinder;

  @Override public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.fragment_pager, container, false);
    unbinder = ButterKnife.bind(this , view);
    adapter = new PagerAdapter(getFragmentManager());
    mViewPager.setOffscreenPageLimit(PAGE_GOUNT - 1);
    mViewPager.setAdapter(adapter);
    mTabs.setViewPager(mViewPager);
    ((MainActivity) getActivity()).setSupportActionBar(mToolbar);
    return view;
  }

  @Override public void onDestroyView() {
    super.onDestroyView();
    unbinder.unbind();
  }


  class PagerAdapter extends FragmentPagerAdapter {
    public PagerAdapter(FragmentManager fm) {
      super(fm);
    }

    @Override
    public Fragment getItem(int position) {
      return MoviesFragment.newInstance(position);
    }

    @Override
    public int getCount() {
      return PAGE_GOUNT;
    }


    @Override
    public CharSequence getPageTitle(int position) {
      if (position == POSITION_POPULAR) {
        return popular;
      } else if (position == POSITION_RATED) {
        return rated;
      } else  {
        return favorites;
      }
    }
  }
}

File: app/src/main/java/com/leodeleon/popmovies/feature/view/MoviePagerFragment.kt
package com.leodeleon.popmovies.feature.view

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.leodeleon.popmovies.R
import com.leodeleon.popmovies.feature.MainActivity
import com.leodeleon.popmovies.util.inflate
import kotlinx.android.synthetic.main.fragment_pager.tabs
import kotlinx.android.synthetic.main.fragment_pager.viewpager
import kotlinx.android.synthetic.main.toolbar.toolbar

class MoviePagerFragment : Fragment() {

  private lateinit var adapter: PagerAdapter

  companion object {
    private val PAGE_GOUNT = 3
  }

  override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
    return container?.inflate(R.layout.fragment_pager, false)
  }

  override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    adapter = PagerAdapter(fragmentManager)
    viewpager.offscreenPageLimit = PAGE_GOUNT - 1
    viewpager.adapter = adapter
    tabs.setViewPager(viewpager)
    (activity as MainActivity).setSupportActionBar(toolbar)
  }

  internal inner class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
      return MoviesFragment.newInstance(position)
    }

    override fun getCount(): Int {
      return PAGE_GOUNT
    }

    override fun getPageTitle(position: Int): CharSequence {
      return when (position) {
        MoviesFragment.POSITION_POPULAR -> context.getString(R.string.popular)
        MoviesFragment.POSITION_RATED -> context.getString(R.string.sort_rated)
        else -> context.getString(R.string.favorites)
      }
    }
  }
}
